### PR TITLE
Refactor for cross database engine support

### PR DESCRIPTION
**Pull request by sarah-tuva**
## Describe your changes
Added functionality to support running in BigQuery in addition to Snowflake and Redshift. I also added primary key tests on fields used for snapshots.

## Validation & Testing
I have tested this in BigQuery, Redshift, and two different Snowflake accounts.

## Checklist before requesting a review
- [X] I have performed a self-review of my code
- [X] My code follows style guidelines
- [X] I have commented my code as necessary
- [X] I have made corresponding changes to documentation
- [X] My code runs without errors
- [X] I have implemented primary key tests
- [X] Every model has a corresponding yml file
- [X] I have generated the DAG locally with dbt and updated the files in the `docs/` path

### CODE REVIEW

File: macros/current_date_or_timestamp.sql [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{#-
+    Returns current date or timestamp in supplied date format (defaults to date)
+    depending on adapter.
+-#}
+
+{%- macro current_date_or_timestamp(date_format='date') -%}
```
Comment (sarah-tuva) [inline macros/current_date_or_timestamp.sql:6]: This macro follows the pattern set by dbt_utils for cross-database support.

File: models/claims/claim_summary.sql [hunk 2]
```diff
@@ -121,7 +121,7 @@ sum_medical_claim_detail as (
 
 ),
 
-add_totals_eligibility_detail as (
+add_denominator_eligibility_detail as (
```
Comment (sarah-tuva) [inline models/claims/claim_summary.sql:124]: Splitting the logic up was necessary as BigQuery doesn't allow you to reference a newly defined column in the same CTE. Both Snowflake and Redshift allow you to do this.

File: models/claims/claim_summary.sql [hunk 3]
```diff
@@ -159,15 +154,36 @@ add_totals_medical_claim_detail as (
...
 ),
 
+add_totals_eligibility_detail as (
+
+    select
+          table_name
+        , test_name
+        , test_fail_numerator
+        , test_fail_denominator
+        , (round(test_fail_numerator / test_fail_denominator, 4)
```
Comment (sarah-tuva) [inline models/claims/claim_summary.sql:168]: Using `round()` and specifying the precision. BigQuery doesn't allow you to cast as a decimal with precision.

File: seeds/seeds.yml [hunk 4]
```diff
@@ -6,7 +6,10 @@ seeds:
       database: "{{ var('output_database') }}"
       schema:  "{{ var('terminology_schema') }}"
       column_types:
-        code: varchar
+        code: |
```
Comment (sarah-tuva) [inline seeds/seeds.yml:9]: I found a limitation with the seeds config. It doesn't support using macros. Using this jinja instead. It's not pretty but it does work. I'm hoping dbt will add this functionality in a future release.

**Review by aneiderhiser — APPROVED** [review]

**Pull request merged**